easybuild/framework/easyconfig/tweak.py

```diff
@@ -615,3 +616,67 @@ def obtain_ec_for(specs, paths, fp=None):
         return res
     else:
         raise EasyBuildError("No easyconfig found for requested software, and also failed to generate one.")
+
+
+def compare_toolchain_specs(source_tc_spec, target_tc_spec):
+    """
+    Compare whether a source and target toolchain have compatible characteristics
+
+    :param source_tc_spec: specs of source toolchain
+    :param target_tc_spec: specs of target toolchain
+    """
+    can_map = True
+    # Check they have same capabilities
+    for key in ['compiler_family', 'mpi_family', 'blas_family', 'lapack_family', 'cuda']:
+        if target_tc_spec[key] is None and source_tc_spec[key] is not None:
+            can_map = False
+            break
+
+    return can_map
+
+
+def match_minimum_tc_specs(source_tc_spec, target_tc_hierarchy):
+    """
+    Match a source toolchain spec to the minimal corresponding toolchain in a target hierarchy
+
+    :param source_tc_spec: specs of source toolchain
+    :param target_tc_hierarchy: hierarchy of specs for target toolchain
+    """
+    minimal_matching_toolchain = {}
+    target_compiler_family = ''
+    # Do a complete loop so we always end up with the minimal value in the hierarchy
+    # hierarchy is given from lowest to highest, so need to reverse the order in the list
+    for target_tc_spec in reversed(target_tc_hierarchy):
+        if compare_toolchain_specs(source_tc_spec, target_tc_spec):
+            # GCCcore has compiler capabilities but should only be used if the original toolchain was also GCCcore
+            if (source_tc_spec['name'] != 'GCCcore' and target_tc_spec['name'] != 'GCCcore')\
```

Comment:
I don't like the hardcoded `'GCCcore'` but I also don't know where I can inherit I that from